products/views.py

```diff
@@ -1,9 +1,58 @@
-import json
+from itertools             import chain
+from django.db.models.expressions import OuterRef
+from django.db.models.fields import CharField
+from django.db.models.query_utils import subclasses
 
-from django.http  import JsonResponse
-from django.views import View
+from django.views          import View
+from django.http           import JsonResponse
+from django.core.paginator import EmptyPage
+from django.db.models      import Q, When, Case, Exists,Value
 
-from .models      import Product, ProductImage, ProductInfo, ImageType,SubCategory
+from .models import Category, SubCategory, Product, ProductInfo, ProductImage, ImageType
+
+class CategoryListView(View):
+    def get(self, request):
+        categories = Category.objects.all()
+        result     = {}
+
+        for category in categories:
+            result[category.english_name] = [sub.english_name for sub in category.sub_category_set.all()]
+
+        return JsonResponse(result, status=200)
+
+class ProductListView(View):
+    def get(self, request):
+        offset        = int(request.GET.get('offset',0))
+        limit         = int(request.GET.get('limit',6))
+        category      = request.GET.get('category')
+        subcategories = request.GET.get('subcategory')
+
+        if subcategories:
+            q = Q(sub_category__english_name=subcategories)
+            print(q)
+        elif category:
+            q = Q(sub_category__category__english_name=category)
+        else:
+            q = Q()
```

Comment:
```suggestion
        q = Q()
        if subcategories:
            q &= Q(sub_category__english_name=subcategories)
            
        if category:
            q &= Q(sub_category__category__english_name=category)
      
```
- print 제거해 주세요.
- 위와 같이 수정하면 좋을 것 같습니다.

Reply:
수정 완료했습니다!